core-bundle/src/Resources/contao/library/Contao/System.php

```diff
@@ -536,6 +536,30 @@ public static function loadLanguageFile($strName, $strLanguage=null, $blnNoCache
 			$GLOBALS['TL_LANG']['MSC']['textDirection'] = (\ResourceBundle::create($strLanguage, 'ICUDATA', true)['layout']['characters'] ?? null) === 'right-to-left' ? 'rtl' : 'ltr';
 		}
 
+		// Backwards compatibility
+		if ('languages' === $strName)
+		{
+			foreach (self::getContainer()->get(Locales::class)->getLocales($strLanguage) as $strLocale => $strLabel)
+			{
+				if (!isset($GLOBALS['TL_LANG']['LNG'][$strLocale]))
```

Comment:
This will not work if a different language is loaded. I think you should move this before loading the actual language file and not check for `$GLOBALS` existence.

Reply:
Fixed in 6629909f483ebfab00d6d189128f95d411c6ed86